### PR TITLE
Audio merge with jaffree

### LINKED ISSUE #53: something is wrong with audio length

### LINKED ISSUE #91: supress logging by jaffree and ffmpeg

### PR DESCRIPTION
**Pull request by mysterix5**
finally i found a way to merge mp3 data properly

- jaffree is used now which is a fork of ffmpeg java wrapper which uses ffmpeg
- additional ffmpeg heroku buildpack is needed
- closes #53 
- closes #91

### CODE REVIEW

**Comment by mysterix5** [comment]
ffmpeg is missing in test on github. mock or install?

File: backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudRepositoryTest.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ void find() throws IOException {
         String url = "https://cloud.com/";
         CloudRepository cloudRepository = new CloudRepository(mockedSardine, url);
 
-        var obFile = new File("src/test/resources/cloud_storage/ob.mp3");
+        var obFile = new File("src/test/resources/cloud_storage/eins.mp3");
         var obStream = new FileInputStream(obFile);
 
         Mockito.when(mockedSardine.get(url + "ob.mp3")).thenReturn(obStream);
```
Comment (NicolaiKopka) [inline backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudRepositoryTest.java:25]: Interesting, what is mockedSardine ?

File: backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudServiceTest.java [hunk 2]
```diff
@@ -17,26 +17,24 @@
...
-            var obDasBytes = obDasStream.readAllBytes();
+            byte[] mergedAudio = cloudService.loadMultipleMp3FromCloudAndMerge(List.of("eins.mp3", "zwei.mp3"));
 
-            AudioInputStream mergedAudio = cloudService.loadMultipleAudioFromCloudAndMerge(List.of("ob.mp3", "das.mp3"));
-
-            assertThat(mergedAudio.readAllBytes()).isEqualTo(obDasBytes);
+            assertThat(mergedAudio.length).isEqualTo(einszweiStream.readAllBytes().length);
         } catch (IOException e) {
             throw new RuntimeException();
         }
```
Comment (NicolaiKopka) [inline backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudServiceTest.java:40]: Now with this try catch, wouldn't your test still be green if the Exception is thrown ?

Comment (NicolaiKopka) [inline backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudServiceTest.java:40]: Couldn't this be two tests ? One that is definetely succeeding and one that tests, that the right exception is actually thrown ?

**Review by NicolaiKopka — APPROVED** [review]
Interesting stuff. Will definetely need to copy your error message handling with VoverErrorDTO ^^

File: backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudRepositoryTest.java (hunk 1 above)
Comment (mysterix5) [reply backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudRepositoryTest.java:25]: Sardine is a library to send webdav requests very easily. I need it to access my cloud storage via webdav

File: backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudServiceTest.java (hunk 2 above)
Comment (mysterix5) [reply backend/src/test/java/com/github/mysterix5/vover/cloud_storage/CloudServiceTest.java:40]: the IOException throw is not controlled by my code so it is not that interesting for me. I make sure to catch it in productive code anyway and i actually change this test in the next PR. But generally: yes, i have to be careful here

**Pull request merged**